contracts/base/ModuleManager.sol

```diff
@@ -40,6 +40,44 @@ abstract contract ModuleManager is SelfAuthorized, Executor, GuardManager {
         }
     }
 
+    /**
+     * @notice Runs pre-execution checks for module transactions if a guard is enabled.
+     * @param to Target address of module transaction.
+     * @param value Ether value of module transaction.
+     * @param data Data payload of module transaction.
+     * @param operation Operation type of module transaction.
+     * @param guard Guard to be used for checking.
+     * @return guardHash Hash returned from the guard tx check.
+     */
+    function runPreExecutionChecks(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        address guard
+    ) internal returns (bytes32 guardHash) {
+        // Only whitelisted modules are allowed.
+        require(msg.sender != SENTINEL_MODULES && modules[msg.sender] != address(0), "GS104");
+
+        if (guard != address(0)) {
+            guardHash = Guard(guard).checkModuleTransaction(to, value, data, operation, msg.sender);
+        }
+    }
+
+    /**
+     * @notice Runs post-execution checks for module transactions if a guard is enabled along with Event emission.
+     * @param guardHash Hash returned from the guard during pre execution check.
+     * @param success Boolean flag indicating if the call succeeded.
+     * @param guard Guard to be used for checking.
+     */
+    function postExecutionChecksWithEventEmissions(bytes32 guardHash, bool success, address guard) internal {
```

Comment:
naming-nano-nit:

I would just call this preModuleExecution and postModuleExecution, I don't think you need to add the "with event emissions"; it feels a bit overly verbose and captured by "this is what we run before and after we execute module transactions".

Just my 2cents - would be good to see how others feel about this as well. I don't feel particularly strongly about the naming, so feel free to override my suggestion depending on how the wider team feels.

Reply:
A shorter name is good 👍🏾 

Done 👍🏾